File: tournaments/dfsComponentSize/dfsComponentSize.js
function dfsComponentSize(matrix, vertex) {
    const visited = Array(matrix.length).fill(0);
    let result = 0;

    const dfs = function(index) {
        visited[index] = 1;
        ++result;

        for (let i = 0; i in matrix; ++i) {
            if (matrix[index][i] && !visited[i]) {
                dfs(i);
            }
        }
    };

    dfs(vertex);

    return result;
}

File: tournaments/dfsComponentSize/dfsComponentSize.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub dfsComponentSize {
    my ($matrix, $vertex) = @_;
    my $result = 1;
    my @queue = ($vertex);
    my %visited;

    $visited{$vertex} = 1;

    while(@queue) {
        my $n = shift(@queue);
        for (my $i = 0; @{@$matrix[$n]} > $i; ++$i) {
            if (@{@$matrix[$n]}[$i] & !$visited{$i}){
                ++$result;
                push(@queue, $i);
                $visited{$i} = 1;
            }
        }
    }

    return $result;
}
